### DISCUSSION TITLE
Filtering by boolean fields, typically filtering where a boolean field is false, fails and returns no fields.

### DISCUSSION THREAD

**Discussion by 1Ghasthunter1** [prisma/prisma#25879] [Q&A]
### Bug description

If we have a model like the following:

```prisma
model User {
  id         String        @id @default(auto()) @map("_id") @db.ObjectId
  name       String
  hidden     Boolean       @default(false)
}
```

Doing any query like 

```ts
await prisma.user.findMany({
  where: {
    hidden: false, //Or, even doing hidden {not: true} fails
  }
})
```

will return an empty array, `[]`, even if rows have `false` values for the `hidden` field.

### How to reproduce

1. Spin up a NextJS application
2. Make a DB with MongoDB, add the user model and create a few entries with hidden being true and false in some
3. Try to query and see what happens 

### Prisma information

```prisma
model User {
  id         String        @id @default(auto()) @map("_id") @db.ObjectId
  name       String
  hidden     Boolean       @default(false)
}```

```ts
await prisma.user.findMany({
  where: {
    hidden: false, //Or, even doing hidden {not: true} fails
  }
})
```

### Environment & setup

- OS: Mac OS-->
- Database: MongoDB
- Node.js version: v22.5.1

### Prisma Version

```
prisma                  : 6.0.1
@prisma/client          : 6.0.1
```

**Comment by ludralph** [comment]
Hi @1Ghasthunter1 

Using the schema you provided, I created records shown below
```
[
  { id: '67615c9a7e3b813d467dd09f', name: 'Alice', hidden: true },
  { id: '67615ca49bb5ed43e6a4c038', name: 'Alice', hidden: false },
  { id: '67615cad111710650cd52a76', name: 'John', hidden: false }
]
```
On performing the query
```js
 const res = await prisma.user.findMany({
    where: {
       hidden: { not: true } 
    }
  })
```
or this other query
```js
 const res = await prisma.user.findMany({
    where: {
      hidden: false, //Or, even doing hidden {not: true} fails
    }
  })
```
I get the correct result returned
```
[
  { id: '67615ca49bb5ed43e6a4c038', name: 'Alice', hidden: false },
  { id: '67615cad111710650cd52a76', name: 'John', hidden: false }
]
```
Can you enable [debugging](https://www.prisma.io/docs/orm/prisma-client/debugging-and-troubleshooting/debugging) and recheck your query?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
